spec/unit/functions/logging_spec.rb

```diff
@@ -6,6 +6,7 @@
 
   def collect_logs(code)
     Puppet[:code] = code
+    Puppet[:environment_timeout] = 10
```

Comment:
Now that the environment isn't reloaded during the test, I'd expect this to no longer be necessary.

Reply:
Actually the guard/ref counting is done in the `compiler` terminus, but this test creates an instance of `Puppet::Parser::Compiler` directly. So you could keep the timeout, or guard the environment explicitly, something like:

```ruby
  around :each do |example|
    envs = Puppet.lookup(:environments)
    envs.guard(:production)
    begin
      example.run
    ensure
      envs.unguard(:production)
    end
  end
```

Reply:
It didn't go as smoothly as expected so I kept the `timeout` for now 😅 .

Reply:
I think what Josh suggested should work, you just need to register the around block anywhere before the actual test cases